packages/dev/serializers/src/glTF/2.0/glTFMaterialExporter.ts

```diff
@@ -84,240 +64,89 @@ function getFileExtensionFromMimeType(mimeType: ImageMimeType): string {
-         * @param specularPower specular power of standard material
-         * @returns Number representing the roughness value
-         */
-        function solveForRoughness(specularPower: number): number {
-            // Given P0.x = 0, P1.x = 0, P2.x = 0
-            //   x = t * t * t * P3.x
-            //   t = (x / P3.x)^(1/3)
-            const t = Math.pow(specularPower / P3.x, 0.333333);
-            return cubicBezierCurve(t, P0.y, P1.y, P2.y, P3.y);
+function ConvertPixelArrayToFloat32(pixels: ArrayBufferView): Float32Array {
+    if (pixels instanceof Uint8Array) {
+        const length = pixels.length;
+        const buffer = new Float32Array(pixels.length);
+        for (let i = 0; i < length; ++i) {
+            buffer[i] = pixels[i] / 255;
         }
+        return buffer;
+    } else if (pixels instanceof Float32Array) {
+        return pixels;
+    } else {
+        throw new Error("Unsupported pixel format!");
+    }
+}
 
-        const diffuse = babylonStandardMaterial.diffuseColor.toLinearSpace(babylonStandardMaterial.getScene().getEngine().useExactSrgbConversions).scale(0.5);
-        const opacity = babylonStandardMaterial.alpha;
-        const specularPower = Scalar.Clamp(babylonStandardMaterial.specularPower, 0, _GLTFMaterialExporter._MaxSpecularPower);
-
-        const roughness = solveForRoughness(specularPower);
-
-        const glTFPbrMetallicRoughness: IMaterialPbrMetallicRoughness = {
-            baseColorFactor: [diffuse.r, diffuse.g, diffuse.b, opacity],
-            metallicFactor: 0,
-            roughnessFactor: roughness,
-        };
+/**
+ * Utility methods for working with glTF material conversion properties.
+ * @internal
+ */
+export class GLTFMaterialExporter {
+    private _exporter: GLTFExporter;
 
-        return glTFPbrMetallicRoughness;
-    }
+    // Mapping to store textures
+    private _textureMap = new Map<BaseTexture, ITextureInfo>();
 
-    /**
-     * Computes the metallic factor
-     * @param diffuse diffused value
-     * @param specular specular value
-     * @param oneMinusSpecularStrength one minus the specular strength
-     * @returns metallic value
-     */
-    public static _SolveMetallic(diffuse: number, specular: number, oneMinusSpecularStrength: number): number {
-        if (specular < this._DielectricSpecular.r) {
-            this._DielectricSpecular;
-            return 0;
-        }
+    // Mapping of internal textures to images to avoid exporting duplicate images
+    private _internalTextureToImage: { [uniqueId: number]: { [mimeType: string]: Promise<number> } } = {};
 
-        const a = this._DielectricSpecular.r;
-        const b = (diffuse * oneMinusSpecularStrength) / (1.0 - this._DielectricSpecular.r) + specular - 2.0 * this._DielectricSpecular.r;
-        const c = this._DielectricSpecular.r - specular;
-        const D = b * b - 4.0 * a * c;
-        return Scalar.Clamp((-b + Math.sqrt(D)) / (2.0 * a), 0, 1);
+    constructor(exporter: GLTFExporter) {

```

Comment:
If this is just assigning a constructor arg to a readonly property, it can be condensed to:
```ts
constructor(private readonly _exporter: GLTFExporter)
```
with the explicit property removed.